pkg/load/configAgent.go

```diff
@@ -61,11 +64,16 @@ func NewConfigAgent(configPath string, cycle time.Duration) (ConfigAgent, error)
 		return nil, fmt.Errorf("Failed to populate watcher: %v", err)
 	}
 	interrupts.Run(func(ctx context.Context) {
-		reloadWatcher(ctx, configWatcher, configCoalescer, a.configPath)
+		reloadWatcher(ctx, configWatcher, a, configCoalescer, a.configPath)
```

Comment:
Does it make sense to still pass `a.configPath` separately now when full `agent` needs to passed in?